R/provider-any.R

```diff
@@ -20,22 +20,33 @@ chat <- function(
   check_string(name, allow_empty = FALSE)
   pieces <- strsplit(name, "/", fixed = TRUE)[[1]]
 
+  provider <- model <- NULL
+
   if (length(pieces) == 1) {
     provider <- pieces[[1]]
-    model <- NULL
-  } else if (length(pieces) == 2) {
+  } else if (length(pieces) >= 2) {
     provider <- pieces[[1]]
-    model <- pieces[[2]]
-  } else {
+    if (length(pieces) == 2) {
+      model <- pieces[[2]]
+    } else if (provider == "github") {
+      # GitHub model names use the `provider/model` format
+      model <- paste(pieces[-1], collapse = "/")
```

Comment:
What about open router and friends? Do they use this format too? Maybe we should just pass along unconditionally and allow the model to error?

Reply:
Yeah that sounds reasonable. It's how I implemented this first until I saw the `chat()` tests and became conflicted. I'd certainly rather let the API throw than to need to maintain a list of exceptions here.